lib/msf/core/post/windows/process.rb

```diff
@@ -44,20 +44,23 @@ def get_notepad_pathname(bits, windir, client_arch)
   end
 
   #
-  # Injects a reflective DLL in to a process, and executes it.
+  # Injects a reflective DLL into a process, and executes it.
   #
   # @param rdll_path [String] The path to the DLL to inject
   # @param param     [String, Integer, nil] The parameter to pass to the DLL's entry point. If this value is a String
   #   then it will first be written into the process memory and then passed by reference. If the value is an Integer,
   #   then the value will be passed as is. If the value is nil, it'll be passed as a NULL pointer.
-  # @param pid       [Integer] The process ID to inject to, if unspecified, a new instance of notepad.exe will be
-  #   launched to host the injected DLL.
+  # @param pid       [Integer] The process ID to inject to, if unspecified, a new instance of a random EXE from the
+  #   process_list array will be launched to host the injected DLL.
   def execute_dll(rdll_path, param=nil, pid=nil)
+    process_list = ['sigverif', 'netsh', 'nslookup', 'winver', 'nbtstat -r 200', 'dxpserver', 'sndvol', 'netstat 200']
```

Comment:
How did we decide on these processes? It seems equally unlikely that winver, netsh, nslookup would spawn subprocesses so it kind of seems like we'd just be substituting notepad for a series of others that would be easily identified as malicious through that type of behavior.

Reply:
These were randomly selected from the System32 directory based on commands that might be commonly executed plus ones that might be lesser known.

Reply:
This list of processes has now been improved on with https://github.com/rapid7/metasploit-framework/pull/15384/commits/1d54a2d854a1a5dc08de93ad4b98acc53cdd64d7 to now focus on those processes that would legitimately spawn subprocesses under various conditions.